src/types.ts

```diff
@@ -0,0 +1,128 @@
+/**
+ * @fileoverview Additional types for this package.
+ * @author Nicholas C. Zakas
+ */
+
+//------------------------------------------------------------------------------
+// Imports
+//------------------------------------------------------------------------------
+
+import type {
+	RuleVisitor,
+	TextSourceCode,
+	Language,
+	LanguageOptions,
+	RuleDefinition,
+} from "@eslint/core";
```

Comment:
At this point I honestly don't understand when types are supposed to be a dev dependency vs. a runtime dependency. What would you do?

Reply:
If references to `@eslint/core` are in the distributed `.d.ts` files, then I believe `@eslint/core` will need to be some kind of a runtime dependency. Otherwise end-users might import types from this package and get type errors on `@eslint/core` not being found.

It does feel weird making a types-only package be a runtime dependency. But 🤷 without first-party support from package managers on delineating "runtime" vs. "types-only" dependencies, it's all "runtime".